src/test/miniscript_tests.cpp

```diff
@@ -116,7 +116,7 @@ struct KeyConverter {
 //! Singleton instance of KeyConverter.
 const KeyConverter CONVERTER{};
 
-using miniscript::operator"" _mst;
+using miniscript::operator"" _mst; // NOLINT(misc-unused-using-decls)
```

Comment:
Is there an upstream bug report?

Reply:
Looks like there is:
https://github.com/llvm/llvm-project/issues/53444
https://github.com/llvm/llvm-project/issues/55095

Seems like the false-positive may have begun appearing from LLVM 14.0.0. I haven't yet tested if it's fixed in 15.0.0.

Reply:
Looks like this passes with clang-13:

```
clang-tidy-13 --use-color -p=/bitcoin-core/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu /bitcoin-core/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/miniscript_tests.cpp
651 warnings generated.
1302 warnings generated.
Suppressed 1302 warnings (1302 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

```diff
diff --git a/ci/test/04_install.sh b/ci/test/04_install.sh
index 453a34ca7..064bee2ca 100755
--- a/ci/test/04_install.sh
+++ b/ci/test/04_install.sh
@@ -115,8 +115,8 @@ if [[ "${RUN_TIDY}" == "true" ]]; then
   export DIR_IWYU="${BASE_SCRATCH_DIR}/iwyu"
   if [ ! -d "${DIR_IWYU}" ]; then
     CI_EXEC "mkdir -p ${DIR_IWYU}/build/"
-    CI_EXEC "git clone --depth=1 https://github.com/include-what-you-use/include-what-you-use -b clang_14 ${DIR_IWYU}/include-what-you-use"
-    CI_EXEC "cd ${DIR_IWYU}/build && cmake -G 'Unix Makefiles' -DCMAKE_PREFIX_PATH=/usr/lib/llvm-14 ../include-what-you-use"
+    CI_EXEC "git clone --depth=1 https://github.com/include-what-you-use/include-what-you-use -b clang_13 ${DIR_IWYU}/include-what-you-use"
+    CI_EXEC "cd ${DIR_IWYU}/build && cmake -G 'Unix Makefiles' -DCMAKE_PREFIX_PATH=/usr/lib/llvm-13 ../include-what-you-use"
     CI_EXEC "cd ${DIR_IWYU}/build && make install $MAKEJOBS"
   fi
 fi
diff --git a/src/test/miniscript_tests.cpp b/src/test/miniscript_tests.cpp
index 1679a5f96..3877fea90 100644
--- a/src/test/miniscript_tests.cpp
+++ b/src/test/miniscript_tests.cpp
@@ -116,7 +116,7 @@ struct KeyConverter {
 //! Singleton instance of KeyConverter.
 const KeyConverter CONVERTER{};
 
-using miniscript::operator"" _mst; // NOLINT(misc-unused-using-decls)
+using miniscript::operator"" _mst;
 
 enum TestMode : int {
     TESTMODE_INVALID = 0,
```

And switch jammy for bookworm, see

https://packages.debian.org/bookworm/clang

Reply:
or alternatively add a link to the bug report in this line

Reply:
> or alternatively add a link to the bug report in this line

I'd rather not revert to using LLVM / clang 13, so have added a link to the upstream report.